### PR TITLE
KVStore: Improve codes for read index worker

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: ref #8864

Problem Summary:

1. Group the impl codes in ReadIndexWorker by class name.
2. Record reuse of read index
3. Some read index result impromenet

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by CalvinNeo** [comment]
image
image

File: dbms/src/Storages/KVStore/Read/LearnerReadWorker.cpp [hunk 1]
```diff
@@ -256,10 +260,18 @@ void LearnerReadWorker::recordReadIndexError(
                     region_id);
                 region_status = RegionException::RegionReadStatus::KEY_NOT_IN_REGION;
             }
+            else if (region_error.has_server_is_busy())
+            {
+                // 1. From TiKV
+                // 2. Read index request timeout
+                GET_METRIC(tiflash_raft_learner_read_failures_count, type_read_index_timeout).Increment();
+                LOG_DEBUG(log, "meet abnormal region error {}, [region_id={}]", resp.ShortDebugString(), region_id);
```
Comment (JaySon-Huang) [inline dbms/src/Storages/KVStore/Read/LearnerReadWorker.cpp:268]: nit: maybe we need INFO level when these region_error happens?

**Review by JaySon-Huang — APPROVED** [review]
LGTM with a nit comment

File: dbms/src/Storages/KVStore/Read/LearnerReadWorker.cpp (hunk 1 above)
Comment (CalvinNeo) [reply dbms/src/Storages/KVStore/Read/LearnerReadWorker.cpp:268]: No, this error is as expected. A read index request is safe to be responded with timeout.

**Pull request merged**